lib/hackney/notification/request_precompiled_letter_state.rb

```diff
@@ -0,0 +1,22 @@
+module Hackney
+  module Notification
+    class RequestPrecompiledLetterState < BaseManualGateway
+      def execute(message_id:)
+        response =
+          notification_gateway.precompiled_letter_state(
+            message_id: message_id
+          )
+
+        update_document(message_id: message_id, status: response[:status])
+        response
+      end
+
+      def update_document(message_id:, status:)
+        # document_store = Hackney::Cloud::Document
```

Comment:
I think you might need this here..
```suggestion
        document_store = Hackney::Cloud::Document
```